doc/extending/extending_theano_gpu.txt

```diff
@@ -0,0 +1,80 @@
+
+.. _extending_theano_gpu:
+
+==============================
+Extending Theano with a GPU Op
+==============================
+
+This tutorial covers how to extend Theano with an op that offers a GPU
+implementation.  It assumes you are familiar with how to write new
+Theano ops.  If that is not the case you should probably follow the
+:ref:`extending_theano` and :ref:`extending_theano_c` section before
+continuing on.
+
+Writing a new GPU op can be done in python for some simple tasks, but
+will usually done in C to access the complete API and avoid paying the
+overhead of a python function call.
+
+Dealing with the context
+========================
+
+One of the major differences with GPU ops is that they require a
+context (a.k.a. device) to execute.  Most of the time you can infer
+the context to run on from your inputs.  There is a way for the user
+to transfer things between contexts and to tag certain varaibles for
+transfer.  It might also be the case that your inputs are not all from
+the same context and you would have to choose which one to run on.
```

Comment:
I see that you continue on that bellow. I think we should document somewhere how we infer the context by default. But I'm not sure where. I think it should be in the use of the new back-end and we should link to it from here.